### PR TITLE
Add properties to alert evidence

### PR DESCRIPTION
**Pull request by OvedHagaiHaruvi**
Adding 3 new properties to alert evidences - 

DetailedRoles - Detailed description of the entity role/s in an alert. Values are free-form.
DeviceEvidence.IpInterfaces - Ip interfaces of the device during the time of the alert.
RegistryValue.MdeDeviceId - A unique identifier assigned to a device by Microsoft Defender for Endpoint.

### CODE REVIEW

**Comment by Danielabom** [comment]
Contacted author to get info.

**Comment by OvedHagaiHaruvi** [comment]
Answered offline .
This is the PR for the CSDL - [Pull request 8073753: Add Properties to alertEvidences - Repos (azure.com)](https://dev.azure.com/msazure/One/_git/AD-AggregatorService-Workloads/pullrequest/8073753?_a=files&path=/Workloads/Microsoft.M365.Defender/override/schema-Prod-beta.csdl)

This is the updated api.md - [api.md - Repos (visualstudio.com)](https://microsoftgraph.visualstudio.com/onboarding/_git/onboarding?path=/reviews/11514-add-m365-alerts-api/api.md&version=GB16671-add-new-properties-to-evidence-in-alerts-v2&_a=preview) (already merged changes)

[API Review 16671](https://microsoftgraph.visualstudio.com/onboarding/_workitems/edit/16671): Add new properties to evidence in alerts_v2 (Host.IpInterface, RegistryValue.MdeDeviceId, BaseEvidence.DetailedRoles)

 

The purpose is adding 3 properties to alert evidence schema - 

DetailedRoles - for each evidence, free texts that give context regarding the evidence role in the alert.
IpInterfaces in Device evidence - list of ip interfaces the device had during alert time
Host for registry value evidence - (this is a bug that we didn't include it until now) - Host that the registry value was seen on.

File: changelog/Microsoft.M365.Defender.json [hunk 1]
```diff
@@ -893,6 +893,74 @@
...
+					"ChangeType": "Addition",
+					"Description": "Added **ipInterfaces** property to [deviceEvidence](https://learn.microsoft.com/en-us/graph/api/resources/security-deviceevidence?view=graph-rest-1.0#properties).",
+					"Target": "deviceevidence"
+				},
+				{
+					"Id": "b3449cc1-39ed-4498-80ba-7edb3402101e",
+					"ApiChange": "Property",
+					"ChangedApiName": "mdeDeviceId",
+					"ChangeType": "Addition",
+					"Description": "Added **mdeDeviceId** property to [registryValueEvidence](https://learn.microsoft.com/en-us/graph/api/resources/security-registryvalueevidence?view=graph-rest-1.0#properties).",
```
Comment (Danielabom) [inline changelog/Microsoft.M365.Defender.json:920]:
```suggestion
					"Description": "Added **mdeDeviceId** property to the [registryValueEvidence](https://learn.microsoft.com/en-us/graph/api/resources/security-registryvalueevidence?view=graph-rest-1.0) resource.",
```

File: changelog/Microsoft.M365.Defender.json [hunk 2]
```diff
@@ -893,6 +893,74 @@
...
+					"ChangeType": "Addition",
+                    "Description": "Added **detailedRoles** property to [alertEvidence](https://learn.microsoft.com/en-us/graph/api/resources/security-alertevidence?view=graph-rest-1.0#properties).",
+                    "Target": "alertEvidence"
+				},
+				{
+					"Id": "b3449cc1-39ed-4498-80ba-7edb3402101e",
+					"ApiChange": "Property",
+					"ChangedApiName": "ipInterfaces",
+					"ChangeType": "Addition",
+					"Description": "Added **ipInterfaces** property to [deviceEvidence](https://learn.microsoft.com/en-us/graph/api/resources/security-deviceevidence?view=graph-rest-1.0#properties).",
```
Comment (Danielabom) [inline changelog/Microsoft.M365.Defender.json:912]:
```suggestion
					"Description": "Added **ipInterfaces** property to the [deviceEvidence](https://learn.microsoft.com/en-us/graph/api/resources/security-deviceevidence?view=graph-rest-1.0) resource.",
```

File: changelog/Microsoft.M365.Defender.json [hunk 3]
```diff
@@ -893,6 +893,74 @@
...
 			"SubArea": "Alerts and incidents"
+		},
+		{
+			"ChangeList": [
+				{
+					"Id": "b3449cc1-39ed-4498-80ba-7edb3402101e",
+					"ApiChange": "Property",
+					"ChangedApiName": "detailedRoles",
+					"ChangeType": "Addition",
+                    "Description": "Added **detailedRoles** property to [alertEvidence](https://learn.microsoft.com/en-us/graph/api/resources/security-alertevidence?view=graph-rest-1.0#properties).",
```
Comment (Danielabom) [inline changelog/Microsoft.M365.Defender.json:904]:
```suggestion
                    "Description": "Added **detailedRoles** property to the [alertEvidence](https://learn.microsoft.com/en-us/graph/api/resources/security-alertevidence?view=graph-rest-1.0) resource.",
```

File: changelog/Microsoft.M365.Defender.json [hunk 4]
```diff
@@ -893,6 +893,74 @@
...
+					"ChangeType": "Addition",
+					"Description": "Added **ipInterfaces** property to [deviceEvidence](https://learn.microsoft.com/en-us/graph/api/resources/security-deviceevidence?view=graph-rest-1.0#properties).",
+					"Target": "deviceevidence"
+				},
+				{
+					"Id": "39ecd7a5-d43c-4bea-b7ef-53d022554b4e",
+					"ApiChange": "Property",
+					"ChangedApiName": "mdeDeviceId",
+					"ChangeType": "Addition",
+					"Description": "Added **mdeDeviceId** property to [registryValueEvidence](https://learn.microsoft.com/en-us/graph/api/resources/security-registryvalueevidence?view=graph-rest-1.0#properties).",
```
Comment (Danielabom) [inline changelog/Microsoft.M365.Defender.json:954]:
```suggestion
					"Description": "Added **mdeDeviceId** property to the [registryValueEvidence](https://learn.microsoft.com/en-us/graph/api/resources/security-registryvalueevidence?view=graph-rest-1.0) resource.",
```

Comment (Danielabom) [inline changelog/Microsoft.M365.Defender.json:954]:
```suggestion
					"Description": "Added **mdeDeviceId** property to the [registryValueEvidence](https://learn.microsoft.com/en-us/graph/api/resources/security-registryvalueevidence?view=graph-rest-1.0) resource.",
```

File: changelog/Microsoft.M365.Defender.json [hunk 5]
```diff
@@ -893,6 +893,74 @@
...
+					"ChangeType": "Addition",
+                    "Description": "Added **detailedRoles** property to [alertEvidence](https://learn.microsoft.com/en-us/graph/api/resources/security-alertevidence?view=graph-rest-1.0#properties).",
+                    "Target": "alertEvidence"
+				},
+				{
+					"Id": "39ecd7a5-d43c-4bea-b7ef-53d022554b4e",
+					"ApiChange": "Property",
+					"ChangedApiName": "ipInterfaces",
+					"ChangeType": "Addition",
+					"Description": "Added **ipInterfaces** property to [deviceEvidence](https://learn.microsoft.com/en-us/graph/api/resources/security-deviceevidence?view=graph-rest-1.0#properties).",
```
Comment (Danielabom) [inline changelog/Microsoft.M365.Defender.json:946]:
```suggestion
					"Description": "Added the **ipInterfaces** property to the [deviceEvidence](https://learn.microsoft.com/en-us/graph/api/resources/security-deviceevidence?view=graph-rest-1.0) resource.",
```

**Review by Danielabom — APPROVED** [review]
Thank you for working on this PR @OvedHagaiHaruvi . PR approved.

**Comment by Danielabom** [comment]
@OvedHagaiHaruvi, please add the "ready to merge" label as soon as you are ready to publish this content. Thanks!

File: api-reference/beta/resources/security-alertevidence.md [hunk 6]
```diff
@@ -30,8 +30,9 @@ This alert evidence base type and its derived evidence types provide a means to
...
 |remediationStatus|[microsoft.graph.security.evidenceRemediationStatus](#evidenceremediationstatus-values)|Status of the remediation action taken. The possible values are: `none`, `remediated`, `prevented`, `blocked`, `notFound`, `unknownFutureValue`.|
 |remediationStatusDetails|String|Details about the remediation status.|
-|roles|[microsoft.graph.security.evidenceRole](#evidencerole-values) collection|One or more roles that an evidence entity represents in an alert. For example, an IP address that is associated with an attacker has the evidence role `Attacker`.|
-|tags|String collection|Array of custom tags associated with an evidence instance. For example, to denote a group of devices or high value assets.|
+|roles|[microsoft.graph.security.evidenceRole](#evidencerole-values) collection|The role/s that an evidence entity represents in an alert, e.g., an IP address that is associated with an attacker will have the evidence role "Attacker".|
+|detailedRoles|String collection|Detailed description of the entity role/s in an alert. Values are free-form.|
```
Comment (Lauragra) [inline api-reference/beta/resources/security-alertevidence.md:34]: Global: Please alphabetize the properties in the Properties tables.

Comment (OvedHagaiHaruvi) [reply api-reference/beta/resources/security-alertevidence.md]: Fixed

File: api-reference/beta/resources/security-alertevidence.md [hunk 7]
```diff
@@ -28,10 +28,11 @@ This alert evidence base type and its derived evidence types provide a means to
...
 |remediationStatus|[microsoft.graph.security.evidenceRemediationStatus](#evidenceremediationstatus-values)|Status of the remediation action taken. The possible values are: `none`, `remediated`, `prevented`, `blocked`, `notFound`, `unknownFutureValue`.|
 |remediationStatusDetails|String|Details about the remediation status.|
-|roles|[microsoft.graph.security.evidenceRole](#evidencerole-values) collection|One or more roles that an evidence entity represents in an alert. For example, an IP address that is associated with an attacker has the evidence role `Attacker`.|
-|tags|String collection|Array of custom tags associated with an evidence instance. For example, to denote a group of devices or high value assets.|
+|roles|[microsoft.graph.security.evidenceRole](#evidencerole-values) collection|The role/s that an evidence entity represents in an alert, e.g., an IP address that is associated with an attacker will have the evidence role "Attacker".|
```
Comment (Danielabom) [inline api-reference/beta/resources/security-alertevidence.md:34]:
```suggestion
|roles|[microsoft.graph.security.evidenceRole](#evidencerole-values) collection|The role/s that an evidence entity represents in an alert, e.g., an IP address that is associated with an attacker will have the evidence role `Attacker`.|
```

**Comment by OvedHagaiHaruvi** [comment]
@Danielabom 
Please review this PR - https://github.com/microsoftgraph/microsoft-graph-docs/pull/22081
I created the same one since merge with master caused my pc to crash.

**Pull request closed without merging**